app/domain/authentication/authn_jwt/validate_and_decode/fetch_issuer_value.rb

```diff
@@ -0,0 +1,149 @@
+require 'uri'
+
+module Authentication
+  module AuthnJwt
+    module ValidateAndDecode
+      # FetchIssuerValue command class is responsible to fetch the issuer secret value,
+      # in order to validate it later against the JWT token claim
+      FetchIssuerValue ||= CommandClass.new(
+        dependencies: {
+          resource_class: ::Resource,
+          fetch_required_secrets: ::Conjur::FetchRequiredSecrets.new,
+          logger: Rails.logger,
+          uri_class: URI
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+        extend(Forwardable)
+        def_delegators(:@authentication_parameters, :service_id, :authenticator_name, :account)
+
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingIssuerConfigurationValue.new)
+          fetch_issuer_value
+          @logger.debug(LogMessages::Authentication::AuthnJwt::FetchedIssuerValueFromConfiguration.new)
```

Comment:
if this would move to the end of `fetch_issuer_value` (right before you return the value) then the `call` function can be 
```
def call
  @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingIssuerConfigurationValue.new)
  fetch_issuer_value
end

Reply:
i now see that we already log `RetrievedIssuerValue` so the `FetchedIssuerValueFromConfiguration` is redundant